[2023-06-12T12:14:41.819+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: s3_download.Download_s3 manual__2023-06-12T12:14:36.373574+00:00 [queued]>
[2023-06-12T12:14:41.903+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: s3_download.Download_s3 manual__2023-06-12T12:14:36.373574+00:00 [queued]>
[2023-06-12T12:14:41.904+0000] {taskinstance.py:1308} INFO - Starting attempt 1 of 6
[2023-06-12T12:14:42.037+0000] {taskinstance.py:1327} INFO - Executing <Task(PythonOperator): Download_s3> on 2023-06-12 12:14:36.373574+00:00
[2023-06-12T12:14:42.058+0000] {standard_task_runner.py:57} INFO - Started process 3089 to run task
[2023-06-12T12:14:42.076+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 's3_download', 'Download_s3', 'manual__2023-06-12T12:14:36.373574+00:00', '--job-id', '324', '--raw', '--subdir', 'DAGS_FOLDER/download_s3_dag.py', '--cfg-path', '/tmp/tmptsonirty']
[2023-06-12T12:14:42.077+0000] {standard_task_runner.py:85} INFO - Job 324: Subtask Download_s3
[2023-06-12T12:14:42.357+0000] {task_command.py:410} INFO - Running <TaskInstance: s3_download.Download_s3 manual__2023-06-12T12:14:36.373574+00:00 [running]> on host 03cd9a7904e9
[2023-06-12T12:14:42.796+0000] {taskinstance.py:1547} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Raja' AIRFLOW_CTX_DAG_ID='s3_download' AIRFLOW_CTX_TASK_ID='Download_s3' AIRFLOW_CTX_EXECUTION_DATE='2023-06-12T12:14:36.373574+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2023-06-12T12:14:36.373574+00:00'
[2023-06-12T12:14:42.811+0000] {s3.py:971} INFO - This function shadows the 'download_file' method of S3 API, but it is not the same. If you want to use the original method from S3 API, please call 'S3Hook.get_conn().download_file()'
[2023-06-12T12:14:42.811+0000] {s3.py:976} INFO - Downloading source S3 file from Bucket s3firstexample with path main.txt
[2023-06-12T12:14:42.879+0000] {base.py:73} INFO - Using connection ID 'S3_conn' for task execution.
[2023-06-12T12:14:42.897+0000] {connection_wrapper.py:337} INFO - AWS Connection (conn_id='S3_conn', conn_type='aws') credentials retrieved from login and password.
[2023-06-12T12:14:54.442+0000] {taskinstance.py:1824} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 181, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 198, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/download_s3_dag.py", line 15, in download_from_s3
    file_name=hook.download_file(key=key,bucket_name=bucket_name,local_path=local_path)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/amazon/aws/hooks/s3.py", line 111, in wrapper
    return func(*bound_args.args, **bound_args.kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/amazon/aws/hooks/s3.py", line 82, in wrapper
    return func(*bound_args.args, **bound_args.kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/amazon/aws/hooks/s3.py", line 1002, in download_file
    file = NamedTemporaryFile(dir=local_path, prefix="airflow_tmp_", delete=False)  # type: ignore
  File "/usr/local/lib/python3.7/tempfile.py", line 547, in NamedTemporaryFile
    (fd, name) = _mkstemp_inner(dir, prefix, suffix, flags, output_type)
  File "/usr/local/lib/python3.7/tempfile.py", line 258, in _mkstemp_inner
    fd = _os.open(file, flags, 0o600)
NotADirectoryError: [Errno 20] Not a directory: '/opt/airflow/dags/main.txt/airflow_tmp_8siqb7fs'
[2023-06-12T12:14:54.477+0000] {taskinstance.py:1350} INFO - Marking task as UP_FOR_RETRY. dag_id=s3_download, task_id=Download_s3, execution_date=20230612T121436, start_date=20230612T121441, end_date=20230612T121454
[2023-06-12T12:14:54.510+0000] {standard_task_runner.py:109} ERROR - Failed to execute job 324 for task Download_s3 ([Errno 20] Not a directory: '/opt/airflow/dags/main.txt/airflow_tmp_8siqb7fs'; 3089)
[2023-06-12T12:14:54.583+0000] {local_task_job_runner.py:225} INFO - Task exited with return code 1
[2023-06-12T12:14:54.657+0000] {taskinstance.py:2651} INFO - 0 downstream tasks scheduled from follow-on schedule check
